snapshots/manager.go

```diff
@@ -59,12 +46,39 @@ type Manager struct {
 	restoreChunkIndex  uint32
 }
 
+// operation represents a Manager operation. Only one operation can be in progress at a time.
+type operation string
+
+// restoreDone represents the result of a restore operation.
+type restoreDone struct {
+	complete bool  // if true, restore completed successfully (not prematurely)
+	err      error // if non-nil, restore errored
+}
+
+const (
+	opNone     operation = ""
+	opSnapshot operation = "snapshot"
+	opPrune    operation = "prune"
+	opRestore  operation = "restore"
+
+	chunkBufferSize = 4
+
+	snapshotMaxItemSize = int(64e6) // SDK has no key/value size limit, so we set an arbitrary limit
+)
+
+var (
+	ErrOptsZeroSnapshotInterval = errors.New("snaphot-interval must not be 0")
+)
+
 // NewManager creates a new manager.
-func NewManager(store *Store, multistore types.Snapshotter, extensions map[string]types.ExtensionSnapshotter) *Manager {
+func NewManager(store *Store, opts types.SnapshotOptions, multistore types.Snapshotter, extensions map[string]types.ExtensionSnapshotter, logger log.Logger) *Manager {
+	multistore.SetSnapshotInterval(opts.Interval)
```

Comment:
This is a bit of a codesmell IMO -- to mutate a type in another types constructor.

Reply:
moved to be set in the caller, updated unit tests